doc/news/changes/incompatibilities/20170506DavidWells

```diff
@@ -0,0 +1,3 @@
+Removed: The deprecated serial PETSc vector classes have been removed.
```

Comment:
Technically yes. Philosophically, I consider this PR to fix all of the bugs with #4279, which includes 'not adding a changelog entry'.